### DISCUSSION TITLE
How to get a Regression Tree model's Feature Importance into a dataframe?

### DISCUSSION THREAD

**Discussion by lobbie** [pycaret/pycaret#3704] [Q&A]
Hi all,

I used PyCaret's Regression Trees module to train a model.  It was very easy to plot the feature importance plot by `plot_model(best_model_1, plot='feature_all')`.   However, I need to get these important features into a dataframe.  

I tried `print(best_model_1.feature_importances_)` and I can get list of feature importance values. 

I also tried to use `X_train_columns = get_config('X_train').columns`  to get the feature names but I got an error `AttributeError: 'NoneType' object has no attribute 'columns'` .  

I can see my list of features are generated by the feature selection functionality in PyCaret
 `TransformerWrapper(exclude=[],
                                     transformer=SelectFromModel(estimator=LGBMRegressor(),
                                                                 max_features=62,
                                                                 threshold=-inf))),`

I have search this forum including Issues but could not find a solution on how to resolve this.

Can someone advise how do I get this list of important features and their relative importance values into a dataframe?  Obviously and since we are able to plot the feature importance plot, we should be able get the data into a dataframe?

Thanks in advance,
Lobbie

**Comment by celestinoxp** [comment]
@ngupta23

**Comment by moezali1** [comment]
@lobbie I am able to do it successfully using the get_config function. See this colab notebook: https://colab.research.google.com/drive/1HfkQRxpJQrIezVZjMGpcpOnFl99Hj0BN?usp=sharing

If your issue still persist please open a bug report.

![image](https://github.com/pycaret/pycaret/assets/54699234/00822914-738f-4422-81ce-225c3096b8be)
